agent/config/agent_config.go

```diff
@@ -65,6 +65,7 @@ const (
 	ENVOY_LISTENER_DRAINING_ENDPOINT_URL = "/drain_listeners"
 	ENVOY_CONCURRENCY_DEFAULT            = -1  // we will not set concurrency [envoy --concurrency] by default.
 	ENVOY_CONCURRENCY_FOR_RELAY_DEFAULT  = "1" // For relay we are defaulting it to 1
+	DISABLE_HOT_RESTART_DEFAULT          = true
```

Comment:
Can we keep the default as `false` and override it via Environment variable from ECS/Fargate agent only in ServiceConnect? Basically keep the existing behavior in AppMesh as is, until we fully understand the root cause of the issue.

Reply:
That's possible. We need to do additional changes on ECS/Fargate Agent to disable-hot-restart on Relay Agent and changes on ECS Service Connect(ECN) Controller to send additional env variable to update SC task definition.

Or after we understand the root cause we can flip the default back to true.

Reply:
Is the root cause relevant to whether we disable this or not? I'd say its driven by whether we need the feature or not. Root cause is interesting but either this is safe or it isn't.

Reply:
Agree. I guess I'm less worried about the root cause than the change in default behavior in AppMesh case. Do we know if there are any customers that rely on the Hot Restart which will not be able to upgrade without updating their code to enable it again?

Reply:
We don't know if there are any customer that would rely on Hot Restart behavior.

Even if they do the Envoy binary is still capable of doing hot restart. There is an option to compile the binary without hot restart ability which we are not exercising right now. For now we just don't expose the ability to do hot restart for existing customer via the SC Agent GoLang script.

Envoy [document](https://www.envoyproxy.io/docs/envoy/latest/operations/hot_restarter) has info on how to do hot restart. Which recommends using a [hot-restarter.py](https://github.com/envoyproxy/envoy/blob/b19956a88f0d7273b0057037e8956fffa1a6ad76/restarter/hot-restarter.py) script. Since we don't manage the hot restart aspects via the Agent script I guess we are not breaking any customer's current usecase of hot restart.

I also believe it is safe to disable since it is a well supported option and nothing about breaking other normal functionality is called out in original PR https://github.com/envoyproxy/envoy/pull/2576 to add the flag and PR  https://github.com/envoyproxy/envoy/pull/2679 to enable the functionality.